src/corppa/ocr/gvision_ocr.py

```diff
@@ -16,47 +16,99 @@
 
 # Attempt to import Google Cloud Vision Python Client
 try:
-    from google.cloud import vision
+    from google.cloud import vision as google_vision
 except ImportError:
-    vision = None
+    google_vision = None
 
 # Workaround (hopefully temporary) to surpress some logging printed to stderr
 os.environ["GRPC_VERBOSITY"] = "NONE"
 
 
-def image_relpath_generator(image_dir, ext_set, follow_symlinks=True):
+def image_relpath_generator(image_dir, exts, follow_symlinks=True):
     """
     This generator method finds all images in image_dir with file extensions
-    in ext_set. For each of these images, the method yields the relative path
-    with respect to image_dir.
+    in exts (case insensitive). For each of these images, the method yields
+    the relative path with respect to image_dir.
 
     For example, if image_dir = "a/b/c/images" and there are image files at the
     following paths: "a/b/c/images/alpha.jpg", "a/b/c/images/d/beta.jpg"
     The generate will produce these two items: "alpha.jpg" and "d/beta.jpg"
     """
+    # Create lowercase extension set from passed in exts
+    ext_set = {ext.lower() for ext in exts}
+
     # Using pathlib.walk over glob because (1) it allows us to find files with
     # multiple extensions in a single walk of the directory and (2) lets us
     # leverage additional functionality of pathlib.
     for dirpath, dirs, files in image_dir.walk(follow_symlinks=follow_symlinks):
         # Check the files in walked directory
         for file in files:
             ext = os.path.splitext(file)[1]
-            if ext:
+            if ext.lower() in ext_set:
                 filepath = dirpath.joinpath(file)
                 yield filepath.relative_to(image_dir)
         # For future walking, remove hidden directories
         dirs[:] = [d for d in dirs if d[0] != "."]
 
 
-def ocr_images(in_dir, out_dir, ext_list, ocr_limit=0, show_progress=True):
+def ocr_image_via_gvision(gvision_client, input_image, out_txt, out_json):
+    """
+    Perform OCR for input image using the Google Cloud Vision API via the provided client.
+    The plaintext output and json response of the OCR call are written to out_txt and
+    out_json paths respectively.
+    """
+    # Check that Google Cloud Vision Python Client was successfully imported
+    if google_vision is None:
+        print(
+            "Error: Python environment does not contain google-cloud-vision "
+            "package. Switch environments or install package and try again.",
+            file=sys.stderr,
+        )
+        sys.exit(1)
```

Comment:
It looks like this check was copied when you refactored out the method but no longer needed. It will already be checked before this function is called, since the client is now passed in. 
```suggestion
```

Reply:
Although I do see that we call other methods on `google_vision` within this method, so it may be reasonable to check. If you think it's a concern (or removing it messes up any unit test assumptions), then you could just refactor the duplicated check to a method that's called in both places.

Reply:
Sure I can refactor it into a separate method, and have it invoked in both places. I think it's important to ensure the check is done in both places, primarily for any future refactoring since `ocr_image_via_gvision` relies directly on `google_vision` and not just the passed in client.

That said, maybe the `ocr_image_via_gvision` thing should gain state and contain the client itself.... but that might be out-of-scope for the current effort.